shuup/core/models/_orders.py

```diff
@@ -583,8 +586,22 @@ def create_refund(self, refund_data, created_by=None):
                     base_unit_price_value=-amount,
                     quantity=1
                 )
+                percent_refunded = amount / parent_line.price if parent_line else 0
                 refund_lines.append(refund_line)
 
+            # refund taxes
+            # we can only refund taxes if we have the order line since different tax classes can apply to
+            # different lines
+            if refund_line and parent_line:
+                for line_tax in parent_line.taxes.all():
+                    refund_line.taxes.create(
+                        tax=line_tax.tax,
+                        name=_("Refund for %s" % line_tax.name),
+                        amount_value=-line_tax.amount_value * percent_refunded,
+                        base_amount_value=-line_tax.base_amount_value * percent_refunded,
+                        ordering=line_tax.ordering
+                    )
+
```

Comment:
Just want to highlight this bit - note that we can't properly refund taxes when the user refunds by amount without selecting a line first. In the future, we should either remove refunds by amount altogether or we should ensure amounts are associated with line items.